### ISSUE TITLE
Tiltfile: run local command and get exit code

### ISSUE THREAD

**Issue by maiamcc** [windmilleng/tilt#2183]
A few times this week, I've written Tiltfile snippets for ppl that run a local command and check if that command succeeded or failed, e.g.:

```def file_exists(f):
    code = local('test -f {}; echo $?'.format(f))
    # exit code 0 means `test` succeeded, i.e. file exists
    return str(code).strip()=='0'```

Notably, we do NOT want to stop Tiltfile execution if the command fails.

Might be nice to have a builtin for `local_with_exit_code` that will do this for us.

**Comment by dbentley** [comment]
1) I think an optional arg to local would be great
2) testing for files and listing them could use improvement separately

**Comment by maiamcc** [comment]
Yeah I know that @drubin had thoughts about this... Dave were you gonna file an issue for your `list_dir` usecase?

**Comment by drubin** [comment]
@maiamcc  Yes but I forgot, will try to do it later today.

**Comment by nicks** [comment]
Another use-case that came up today was:
- call `local()`
- if `local()` fails, run some additional code (and not fail the entire tiltfile execution)

i wonder if there's some better api here that we could borrow from python where you get both the status code and the output

**Comment by nicks** [comment]
e.g., check the current cluster auth, and if it fails, run these commands to try to get creds
